msu/systems/registry/registry_system.nut

```diff
@@ -34,12 +34,12 @@
 				::logError("Duplicate Mod ID for mod: " + _mod.getID());
 				throw ::MSU.Exception.DuplicateKey(_mod.getID());
 			}
-			if (::mods_getRegisteredMod(_mod.getID()) == null)
+			if (::Hooks.getMod(_mod.getID()) == null)
 			{
 				::logError("Register your mod using the same ID with mod_hooks before creating a ::MSU.Class.Mod");
 				throw ::MSU.Exception.KeyNotFound(_mod.getID());
 			}
-			if (::mods_getRegisteredMod(_mod.getID()).SemVer == null || ::MSU.SemVer.getVersionString(::mods_getRegisteredMod(_mod.getID()).SemVer) != _mod.getVersionString())
+			if (typeof ::Hooks.getMod(_mod.getID()).getVersionString() == "float" || ::Hooks.getMod(_mod.getID()).getVersionString() != _mod.getVersionString())
```

Comment:
This float check won't work since getVersionString is always a string, the second check is sufficient regardless